simulator/risc_v/riscv_instr.cpp

```diff
@@ -175,159 +180,159 @@ template<typename I>
-    {'I', instr_lui,    execute_lui<I>,    OUT_ARITHM, 'U', Imm::LOGIC, Src1::ZERO, Src2::ZERO, Dst::RD,   0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_auipc,  execute_auipc<I>,  OUT_ARITHM, 'U', Imm::LOGIC, Src1::ZERO, Src2::ZERO, Dst::RD,   0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
+    {'I', instr_invalid,    do_nothing<I>,     OUT_ARITHM, ImmediateType::NONE,       Imm::NO,       { Src::ZERO,     Src::ZERO },     { Dst::ZERO },     0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
+    {'I', instr_lui,        execute_lui<I>,    OUT_ARITHM, ImmediateType::U,          Imm::LOGIC,    { Src::ZERO,     Src::ZERO },     { Dst::RD },       0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
+    {'I', instr_auipc,      execute_auipc<I>,  OUT_ARITHM, ImmediateType::U,          Imm::LOGIC,    { Src::ZERO,     Src::ZERO },     { Dst::RD },       0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
     // Jumps and branches
-    {'I', instr_jal,    execute_jal<I>,    OUT_J_JUMP, 'J', Imm::ARITH, Src1::ZERO, Src2::ZERO, Dst::RD,   0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_jalr,   execute_jalr<I>,   OUT_R_JUMP, 'I', Imm::LOGIC, Src1::RS1,  Src2::ZERO, Dst::RD,   0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_beq,    execute_beq<I>,    OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_bne,    execute_bne<I>,    OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_blt,    execute_blt<I>,    OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_bge,    execute_bge<I>,    OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_bltu,   execute_bltu<I>,   OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
-    {'I', instr_bgeu,   execute_bgeu<I>,   OUT_BRANCH, 'B', Imm::ARITH, Src1::RS1,  Src2::RS2,  Dst::ZERO, 0, 32 | 64 | 128}, // NOLINT(hicpp-signed-bitwise) https://bugs.llvm.org/show_bug.cgi?id=44977
```

Comment:
Please roll back this change. The original intention is to use a signle-character identifier since it match RISC-V immediate definitions. Enumeration has been introduced to support compressed instructions and we shoul think how we can roll it back to something more readable.